docs/reference/api/media.md

```diff
@@ -0,0 +1,153 @@
+# Media APIs
+
+
+The `@apostrophecms/attachment` module manages all media file upload. Once uploaded, metadata and organization of image and other files are primarily managed through the respective `@apostrophecms/image` and `@apostrophecms/file` piece types. See the [pieces REST API documentation](./pieces.md) for additional information.
+
+## Endpoints
+
+| Method | Path | Description | Auth required |
+|---------|---------|---------|---------|
+|POST | [`/api/v1/@apostrophecms/attachment/upload`](#post-api-v1-apostrophecms-attachment-upload) | Upload a media file | TRUE |
+|POST | [`/api/v1/@apostrophecms/attachment/crop`](#post-api-v1-apostrophecms-attachment-crop) | Crop an image file | TRUE |
+|GET | [`/api/v1/@apostrophecms/image`](#media-get-request) | Get an image document with a specified ID | FALSE |
+|POST | [`/api/v1/@apostrophecms/image`](#media-post-request) | Insert a new image | TRUE |
+|GET | [`/api/v1/@apostrophecms/file`](#media-get-request) | Get a file document with a specified ID | FALSE |
+|POST | [`/api/v1/@apostrophecms/file`](#media-post-request) | Insert a new file | TRUE |
+
+## `POST: /api/v1/@apostrophecms/attachment/upload`
+
+**Authentication required.**
+
+The `body` of the request should be a `FormData` object.
```

Comment:
This documentation assumes you're working in the browser, which may not be the case. Could be a native app, curl, wget, another server, etc. We need to address what goes over the wire here, not just how to do it with browser APIs.

"The body of the request should use the `multipart/form-data` encoding, with the file itself uploaded under the name `file`."

Your example is great for browser-based uploads but as it's written now it sounds like you can't do it with something else.

Reply:
We should seriously consider including `curl`-based examples, which are common in REST API docs and drive home the universality.

Reply:
I'm all for additional examples. Eventually I'd like to have a tab component to cleanly display multiple example types. [I created a task for this](https://apostrophecms.atlassian.net/browse/A3DOCS-34) (auth required) to follow up.